### PR TITLE
Using execution_environment option in ad_hoc_command module

### LINKED ISSUE #14104: ad_hoc_command: execution_environment option is ignored
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.
- [X] I am **NOT** reporting a (potential) security vulnerability. (These should be emailed to `security@ansible.com` instead.)

### Bug Summary

The `execution_environment` option in `ad_hoc_command` module is ignored and runs with the default EE (`AWX EE (latest)`).

### AWX version

22.3.0

### Select the relevant components

- [ ] UI
- [ ] UI (tech preview)
- [ ] API
- [ ] Docs
- [X] Collection
- [ ] CLI
- [ ] Other

### Installation method

N/A

### Modifications

no

### Ansible version

_No response_

### Operating system

_No response_

### Web browser

_No response_

### Steps to reproduce

Run with `execution_environment` option in the `awx.awx.ad_hoc_command` module. As in the following Playbook.

```yaml
    - name: Ad Hoc command test
      awx.awx.ad_hoc_command:
        inventory: Demo Inventory
        credential: Demo Credential 
        module_name: command
        module_args: echo I <3 Ansible
        execution_environment: my_ee
        wait: true
````

### Expected results

Runs in the execution environment specified by the execution_environment option.

### Actual results

The execution_environment option is ignored and runs with the default EE (`AWX EE (latest)`).

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by akira6592**
##### SUMMARY

Fixes #14104

Using execution_environment option in `ad_hoc_command` module.

##### ISSUE TYPE

 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME

 - Collection
 

##### AWX VERSION

```
awx: 21.5.1.dev1166+g05499536ed.d20230610
```

##### ADDITIONAL INFORMATION

```
N/A
```

### CODE REVIEW

**Comment by fosterseth** [comment]
thanks for this contribution, we'll be taking a look at it soon

File: awx_collection/plugins/modules/ad_hoc_command.py [hunk 1]
```diff
@@ -158,6 +159,8 @@ def main():
     # Attempt to look up the related items the user specified (these will fail the module if not found)
     post_data['inventory'] = module.resolve_name_to_id('inventories', inventory)
     post_data['credential'] = module.resolve_name_to_id('credentials', credential)
+    if execution_environment:
+        post_data['execution_environment'] = module.resolve_name_to_id('execution_environments', execution_environment)
```
Comment (john-westcott-iv) [inline awx_collection/plugins/modules/ad_hoc_command.py:163]: Can we add an test in `awx_collection/tests/integration/targets/ad_hoc_command/tasks/main.yml` for this?

Comment (akira6592) [reply awx_collection/plugins/modules/ad_hoc_command.py:163]: @john-westcott-iv Thank you for reviewing. I added integration test.

**Review by jessicamack — APPROVED** [review]

**Comment by jessicamack** [comment]
@akira6592 Thank you again for the contribution. I've confirmed the new integration test passes. Approving and merging.

**Comment by akira6592** [comment]
@jessicamack Thank you for merge!

**Pull request merged**